content/livesync/mongodb-connector.textile

```diff
@@ -0,0 +1,197 @@
+      "operationType": {
+        "$in": [
+          "update",
+          "insert"
+        ]
+      }
+    }
+  },
+  {
+    "$set": {
+      "_ablyChannel": "myChannelName"
+    }
+  }
+]
+```
+
+The pipeline must set the @_ablyChannel@ field on the change event.
+
+
+h2(#channel-routing). Routing change events to channels
+
+The integration rule will listen to change events from the MongoDB ChangeStream and route change events to Ably channels using the @_ablyChannel@ field set on the change event. You can dynamically route change events to channels using the Pipeline configuration. Ably will publish each change event as an individual message to the associated channel as received from the ChangeStream subscription, without modification.
+
+<aside data-type='important'>
+    You must set the @_ablyChannel@ field on the root of the change event.
+</aside>
+
+```[json]
+// Set the _ablyChannel field statically
+[{ "$set": { "_ablyChannel": "myChannelName" } }]
+
+// Set the _ablyChannel field based on a field of the fullDocument:
+[{ "$set": { "_ablyChannel": "$fullDocument.channel" } }]
+
+// Set the _ablyChannel field dynamically based on the content of the change event:
+[{
+  "$set": {
+    "_ablyChannel": {
+      "$concat": [
+        "todos:","$fullDocument.workspace_id",":user:","$fullDocument.user_id"
+      ]
+    }
+  }
+}]
+
+// Set the _ablyChannel field dynamically by coalescing fullDocument and fullDocumentBeforeChange:
+[{
+  "$set": {
+    "_ablyChannel": {
+      "$ifNull": [
+        "$fullDocument.channel","$fullDocumentBeforeChange.channel"
+      ]
+    }
+  }
+}]
+```
+
+h2(#ably-fields). _ably fields
+
+The connector looks for the following fields:
+
+|_. Field |_. Required |_. Type |_. Description |
+| @_ablyChannel@ | required | string | Used as the channel that the change event will be published to Ably |
+| @_ablyMsgName@ | optional | string | Used as the message name when the change event is published to Ably |
+| @_ablyMsgHeaders@ | optional | object with string keys and values | Used se the messages extras.headers when the change event is published to Ably |
+
+
+h2(#event-schema). Change event schema
+
+The changes events received from MongoDB ChangeStream, after the pipeline is applied, are not modified and are published as messages on the @_ablyChannel@ as-is.
+
+Please refer to the "MongoDB ChangeStream event schema docs":https://www.mongodb.com/docs/v8.0/reference/change-events/ for change event stream types and schema.
+
+h2(#full-document). Full Document
+
+The Full Document configuration defines if the ChangeStream should include the full document. Some change event types (e.g. update) only include the updated fields, and do not include the full document by default.
+
+Access to the full document can be useful for constructing the @_ablyChannel@ field.
+
+The Full Document config operates as follows:
+
+|_. Option |_. Description |
+| off | There will be no "fullDocument" field on the change event |
```

Comment:
We should link to the corresponding docs on mongo's website

Reply:
There's actually really pants docs on this on MongoDB. 

Each SDK has the options that you can pass, and there they sort-of specify the values. But in the change stream docs they go deep on `updateLookup` without talking about the other options from pre-and-post images.